File: fjt.api/models/ReportMaster.js
const Sequelize = require('sequelize');

module.exports = (sequelize) => {
    const ReportMaster = sequelize.define('ReportMaster', {
        id: {
            allowNull: false,
            autoIncrement: true,
            primaryKey: true,
            type: Sequelize.INTEGER
        },
        reportName: {
            type: Sequelize.STRING,
            allowNull: true
        },
        rdlcReportFileName: {
            type: Sequelize.STRING,
            allowNull: true
        },
        reportTitle: {
            type: Sequelize.STRING,
            allowNull: true
        },
        customerID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        partID: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        companyID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        fromDate: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        toDate: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        logicalExpression: {
            type: Sequelize.STRING,
            allowNull: true
        },
        isDeleted: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        createdBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        updatedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        deletedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        withAlternateParts: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        customerSelectType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        partSelectType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        employeeID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        employeeSelectType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        supplierID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        supplierSelectType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        mfgCodeID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        mfgCodeSelectType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        assyID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        assySelectType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        mountingTypeID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        mountingTypeSelectType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        functionalTypeID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        functionalTypeSelectType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        partStatusID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        partStatusSelectType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        workorderID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        workorderSelectType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        operationID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        operationSelectType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        reportCategoryId: {
            type: Sequelize.INTEGER,
            allowNull: false
        },
        reportViewType: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        reportAPI: {
            type: Sequelize.STRING,
            allowNull: true
        },
        isExcel: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        emailTemplete: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        createByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        updateByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        deleteByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        fromTime: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        toTime: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        fileName: {
            allowNull: false,
            type: Sequelize.STRING(255)
        },
        isEndUserReport: {
            allowNull: true,
            type: Sequelize.BOOLEAN
        },
        draftFileName: {
            allowNull: true,
            type: Sequelize.STRING(255)
        },
        radioButtonFilter: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        additionalNotes: {
            type: Sequelize.STRING(2000),
            allowNull: true
        },
        isCSVReport: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        csvReportAPI: {
            type: Sequelize.TEXT,
            allowNull: true
        },
        refReportId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        status: {
            type: Sequelize.STRING(1),
            allowNull: true
        },
        entityId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        editingBy: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        startDesigningDate: {
            type: Sequelize.DATE,
            allowNull: true
        },
        reportGenerationType: {
            type: Sequelize.STRING(1),
            allowNull: true
        },
        reportVersion: {
            type: Sequelize.STRING(255),
            allowNull: true
        },
        isDefaultReport: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        }
    }, {
        tableName: 'reportmaster',
        paranoid: true
    });

    ReportMaster.associate = (models) => {
        ReportMaster.hasMany(models.EmailScheduleMst, {
            foreignKey: 'reportID',
            as: 'EmailScheduleMst'
        });
        ReportMaster.belongsTo(models.Entity, {
            foreignKey: 'entityId',
            as: 'Entity'
        });
        ReportMaster.belongsTo(models.GenericCategory, {
            foreignKey: 'reportCategoryId',
            as: 'GenericCategory'
        });
        ReportMaster.hasMany(models.ReportChangeLogs, {
            foreignKey: 'reportId',
            as: 'Report_Change_Logs'
        });
        ReportMaster.hasMany(models.ReportViewerParameter, {
            foreignKey: 'reportId',
            as: 'ReportViewerParameter'
        });
        ReportMaster.hasMany(models.ReportMasterParameter, {
            foreignKey: 'reportId',
            as: 'ReportMasterParameter'
        });
    };

    return ReportMaster;
};

File: FJT.ReportViewer/MySqlDBModel/reportmaster.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace FJT.ReportViewer.MySqlDBModel
{
    public class reportmaster
    {
        //public reportmaster()
        //{
        //    report_change_logs = new HashSet<report_change_logs>();
        //    reportviewerparameter = new HashSet<reportviewerparameter>();
        //}

        [Key]
        public int id { get; set; }

        [StringLength(255)]
        public string reportName { get; set; }

        [StringLength(255)]
        public string rdlcReportFileName { get; set; }

        [StringLength(255)]
        public string reportTitle { get; set; }

        public int? customerID { get; set; }

        public int? partID { get; set; }

        public int? companyID { get; set; }

        public int? fromDate { get; set; }

        public int? toDate { get; set; }

        public string logicalExpression { get; set; }

        public bool isDeleted { get; set; }        

        public DateTime createdAt { get; set; }

        [StringLength(255)]
        public string createdBy { get; set; }

        public DateTime? updatedAt { get; set; }

        [StringLength(255)]
        public string updatedBy { get; set; }

        public DateTime? deletedAt { get; set; }

        [StringLength(255)]
        public string deletedBy { get; set; }

        public int? withAlternateParts { get; set; }

        public int? customerSelectType { get; set; }

        public int? partSelectType { get; set; }

        public int? employeeID { get; set; }

        public int? employeeSelectType { get; set; }

        public int? supplierID { get; set; }

        public int? supplierSelectType { get; set; }

        public int? mfgCodeID { get; set; }

        public int? mfgCodeSelectType { get; set; }

        public int? assyID { get; set; }

        public int? assySelectType { get; set; }

        public int? mountingTypeID { get; set; }

        public int? mountingTypeSelectType { get; set; }

        public int? functionalTypeID { get; set; }

        public int? functionalTypeSelectType { get; set; }

        public int? partStatusID { get; set; }

        public int? partStatusSelectType { get; set; }

        public int? workorderID { get; set; }

        public int? workorderSelectType { get; set; }

        public int? operationID { get; set; }

        public int? operationSelectType { get; set; }

        public int? reportCategoryId { get; set; }

        public bool? reportViewType { get; set; }

        [StringLength(200)]
        public string reportAPI { get; set; }

        public bool? isExcel { get; set; }

        public int? createByRoleId { get; set; }

        public int? updateByRoleId { get; set; }

        public int? deleteByRoleId { get; set; }

        public int? emailTemplete { get; set; }

        public int? fromTime { get; set; }

        public int? toTime { get; set; }

        [StringLength(255)]
        public string fileName { get; set; }

        public bool? isEndUserReport { get; set; }

        [StringLength(255)]
        public string draftFileName { get; set; }

        public int? radioButtonFilter { get; set; }

        public string additionalNotes { get; set; }

        public bool isCSVReport { get; set; }

        public string csvReportAPI { get; set; }

        public int? refReportId { get; set; }

        [StringLength(1)]
        public string status { get; set; }

        public int? entityId { get; set; }

        public int? editingBy { get; set; }

        public DateTime? startDesigningDate { get; set; }

        public string reportGenerationType { get; set; }

        [StringLength(255)]
        public string reportVersion { get; set; }

        public bool isDefaultReport { get; set; }

        [ForeignKey("entityId")]
        public virtual entity entity { get; set; }

        [ForeignKey("reportCategoryId")]
        public virtual genericcategory genericcategory { get; set; }

        //public virtual ICollection<report_change_logs> report_change_logs { get; set; }
        //public virtual ICollection<reportviewerparameter> reportviewerparameter { get; set; }
    }
}
